### ISSUE TITLE
Multiple Owners check

### ISSUE THREAD

**Issue by micimize** [mszostok/codeowners-validator#70]
CODEOWNERS has a problematic behavior where a rule that specifies a single user will [prevent that user from editing their owned code](https://github.community/t/do-not-require-owner-approval-if-the-pull-request-is-from-an-owner/369). 

It would be nice to validate that there are no rules of the form `/js @myjsguy` to prevent this from happening, maybe via a `multiowners` flag.

**Comment by cktaylor** [comment]
It is mentioned in the linked thread - but to call it out explicitly, another situation to consider (that may be out of scope of this issue though) is that it is equally problematic if the CODEOWNERS rule specifies a single team and that team has a single user.

Given that `codeowners-validator` already has logic to talk to Github for team/org metadata - if it is possible to get team size from that, and if the proposed `multiowners` flag/mode is being added - I just wanted to point out this broader functionality would also be useful.

**Comment by micimize** [comment]
@cktaylor this is true, but a much harder thing to put guard rails on, as team size can change externally and cause issues
